src/application/LegendVSCodeApplication.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  type LegendApplicationConfigurationInput,
+  ApplicationFrameworkProvider,
+  ApplicationStore,
+  ApplicationStoreProvider,
+  assertErrorThrown,
+  BrowserEnvironmentProvider,
+  Core_GraphManagerPreset,
+  Core_LegendApplicationPlugin,
+  QueryBuilder_GraphManagerPreset,
+  QueryBuilder_LegendApplicationPlugin,
+} from '@finos/legend-vscode-extension-dependencies';
+import { useMemo } from 'react';
+import {
+  LegendVSCodeApplicationConfig,
+  type LegendVSCodeApplicationConfigurationData,
+} from './LegendVSCodeApplicationConfig';
+import { LegendVSCodePluginManager } from './LegendVSCodePluginManager';
+import { Core_LegendVSCodeApplicationPlugin } from './Core_LegendVSCodeApplicationPlugin';
+import { postMessage } from '../utils/VsCodeUtils';
+import { QUERY_BUILDER_CONFIG_ERROR } from '../utils/Const';
+
+export const LegendVSCodeApplication = (props: {
+  configData: LegendVSCodeApplicationConfigurationData;
+  children: React.ReactNode;
+}): React.ReactNode => {
+  const { configData, children } = props;
+
+  const applicationStore = useMemo(() => {
+    const input: LegendApplicationConfigurationInput<LegendVSCodeApplicationConfigurationData> =
+      {
+        baseAddress: '',
```

Comment:
`baseAddress` and `versionData` are required by the `LegendApplicationConfigurationInput` interface. I'm not sure they're really applicable for the VS code extension, but I'm open to suggestions of what should be put here.

Reply:
Can the interface be updated to mark these as optional. An/Mauricio can help here.

Reply:
Done